File: Leetcode/257_Binary_Tree_Paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<string> ans;
    
    vector<string> binaryTreePaths(TreeNode* root) {
        if(root == nullptr)
        {
            return ans;
        }   
        if(root->left == nullptr && root->right == nullptr)
        {
            string s = to_string(root->val);
            ans.push_back(s);
        }
        else
        {
            helper(root, to_string(root->val));   
        }
        
        return ans;
    }
    
    void helper(TreeNode* root, string s)
    {
        if(root != nullptr)
        {
            if(root->left == nullptr && root->right == nullptr)
            {
                ans.push_back(s);
            }
            
            if(root->left != nullptr)
            {
                helper(root->left, s + "->" + to_string(root->left->val));
            }
            
            if(root->right != nullptr)
            {
                helper(root->right, s + "->" + to_string(root->right->val));
            }
        }
    }
};

File: Leetcode/0257/257_Binary_Tree_Paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function(root) {
    let result = [];
    let path = "";
    
    helper(root, result, path);
    return result;
};

const helper = function(root, result, path) {
    if (root) {
        path = path === "" ? `${root.val}` : `${path}->${root.val}`;
        if (root.left === null && root.right === null) {
            result.push(path);
            return;
        }
        
        root.left ? helper(root.left, result, path) : null;
        root.right ? helper(root.right, result, path) : null;
    }
}